subprojects/internal-integ-testing/src/main/groovy/org/gradle/integtests/fixtures/console/AbstractConsoleGroupedTaskFunctionalTest.groovy

```diff
@@ -19,36 +19,38 @@ package org.gradle.integtests.fixtures.console
 import org.gradle.api.logging.configuration.ConsoleOutput
 import org.gradle.integtests.fixtures.AbstractIntegrationSpec
 import org.gradle.integtests.fixtures.RichConsoleStyling
+import org.gradle.integtests.fixtures.executer.ConsoleAttachment
+import org.junit.Assume
+import org.junit.runner.RunWith
 
 /**
  * A base class for testing the console.
  */
+@RunWith(ConsoleAttachmentTestRunner.class)
 abstract class AbstractConsoleGroupedTaskFunctionalTest extends AbstractIntegrationSpec implements RichConsoleStyling {
-    protected boolean consoleAttached
-    protected boolean stderrAttached
+    static ConsoleAttachment consoleAttachment
 
     def setup() {
+        // TODO - rich/verbose consoles currently exhibit slightly different behavior than these tests expect when no console is attached
+        Assume.assumeFalse(consoleAttachment == ConsoleAttachment.NOT_ATTACHED && (consoleType == ConsoleOutput.Rich || consoleType == ConsoleOutput.Verbose))
+
+        switch(consoleAttachment) {
+            case ConsoleAttachment.NOT_ATTACHED:
+                break
+            case ConsoleAttachment.ATTACHED:
+                executer.withTestConsoleAttached()
+                break
+            case ConsoleAttachment.ATTACHED_STDOUT_ONLY:
+                executer.withTestConsoleAttachedToStdoutOnly()
```

Comment:
would be simpler if there were a `withConsoleAttachment()` on `GradleExecuter` instead.

Reply:
ah, I see this is in the next PR. Nevermind then.